### PR TITLE
Add centered FFT example to fftshift docs

### LINKED ISSUE #51022: Update fftshift, roll, and ifftshift documentation
## 🚀 Feature
Add `fftshift`, `ifftshift`, and `roll` functions to PyTorch fft as `torch.fft.fftshift`, `torch.fft.ifftshift` and `torch.roll`.

## Motivation

These functions are necessary for proper and easy use of FFT, which has been recently added to PyTorch https://github.com/pytorch/pytorch/issues/42175.

## Pitch

This is already implemented in FastMRI [here](https://github.com/facebookresearch/fastMRI/blob/0f538a353e4a5c855012b8f982f756d489d11462/fastmri/fftc.py#L61). FastMRI is an official project by Facebook Research. Simply copy-pasting the fftc.py file somewhere would be good enough.

## Alternatives

Cannot think of any.

## Additional context

I would also like to request documentation on the proper use of `fftshift`.

As mentioned in this issue https://github.com/numpy/numpy/issues/13442, The proper way to apply the fast Fourier transform is `fftshift(fft(ifftshift(x)))` and the proper way to apply ifft is `fftshift(ifft(ifftshift(X)))`, which is how the function is being used in the FastMRI project. However, many people do not know this. An explanation would be helpful.

cc @jlin27 @mruberry @peterbell10 @walterddr

### PR DESCRIPTION
**Pull request by peterbell10**
Closes #51022

### CODE REVIEW

File: torch/fft/__init__.py [hunk 1]
```diff
@@ -889,10 +893,10 @@
 
     >>> f = torch.fft.fftfreq(4)
     >>> f
-    tensor([ 0.0000,  0.2500,  -0.5000, -0.2500])
+    tensor([ 0.0000,  0.2500, -0.5000, -0.2500])
 
-    >>> torch.fftshift(f)
-    tensor([-0.5000, -0.2500, 0.0000, 0.2500])
+    >>> torch.fft.fftshift(f)
+    tensor([-0.5000, -0.2500,  0.0000,  0.2500])
```
Comment (peterbell10) [inline torch/fft/__init__.py:899]: Found these discrepancies from running the doctests manually. So, it seems fft doctests aren't being run in CI.

File: torch/fft/__init__.py [hunk 2]
```diff
@@ -865,6 +865,10 @@
 Reorders n-dimensional FFT data, as provided by :func:`~torch.fft.fftn`, to have
 negative frequency terms first.
 
+This performs a periodic shift of n-dimensional data such that the origin
+``(0, ..., 0)`` is moved to the center of the tensor. Specifically,
```
Comment (mruberry) [inline torch/fft/__init__.py:869]: Suggestion: 

"Specifically, ..." -> "Specifically, to ..." to make the second sentence parallel to the first.

File: torch/fft/__init__.py (hunk 1 above)
Comment (mruberry) [reply torch/fft/__init__.py:899]: Nice fix.

File: torch/fft/__init__.py [hunk 3]
```diff
@@ -914,6 +918,28 @@
...
+    >>> x_uncentered = torch.fft.ifftshift(x_centered)
+    >>> fft_uncentered = torch.fft.fft(x_uncentered)
+
+    Similarly, we can convert the frequency domain components to centered
+    convention by applying :func:`~torch.fft.fftshift`.
+
+    >>> fft_centered = torch.fft.fftshift(fft_uncentered)
+
+    The inverse transform, from centered Fourier space back to centered spatial
+    data, is simply applying the inverse shifts in reverse order:
```
Comment (mruberry) [inline torch/fft/__init__.py:936]: Suggestion: "is simply applying" -> "can be performed by applying"

**Review by mruberry — APPROVED** [review]
I made a couple suggestions for your review @peterbell10. If they're helpful, great, and if not no worries. Just ping me when you'd like this merged.

**Comment by peterbell10** [comment]
@mruberry I've added your suggestions.

**Pull request closed without merging**